DependencyInjection/Compiler/DescriptionEnhancerPass.php

```diff
@@ -59,12 +70,26 @@ public function process(ContainerBuilder $container)
             ));
         }
 
+        // Remove inactive enhancers from the container.
+        //
+        // Note that previously we relied on the container automatically removing
+        // unreferenced definitions, but this proved to not always be the case. It
+        // is safer to remove them here.
         $inactiveEnhancers = array_diff($enhancerNames, $enabledEnhancers);
```

Comment:
Argh. Forget it. In `master` we remove the inactive enhancers from the container, but we also unset the enhancer from the factory. In removing one I also removed the other and drew the wrong conclusion. All is well.